pkg/graph/resolvers/it_gov_task_statuses.go

```diff
@@ -6,7 +6,24 @@ import "github.com/cmsgov/easi-app/pkg/models"
 
 // IntakeFormStatus calculates the ITGovTaskListStatus of a system intake for the requester view
 func IntakeFormStatus(intake *models.SystemIntake) models.ITGovIntakeFormStatus {
-	return models.ITGISReady
+	if intake.Step != models.SystemIntakeStepINITIALFORM {
+		return models.ITGISCompleted
+	}
+	switch intake.RequestFormState {
+	case models.SIRFSNotStarted:
+		return models.ITGISReady
+
+	case models.SIRFSInProgress:
+		return models.ITGISInProgress
+
+	case models.SIRFSEditsRequested:
+		return models.ITGISEditsRequested
+
+	case models.SIRFSSubmitted:
+		return models.ITGISCompleted
+	default: //This is included to be explicit. This should not technically happen in normal use, but it is technically possible as the type is a type alias for string
```

Comment:
I think I would rather throw an error in the default case if we have to include one to be exhaustive in the switch.

Reply:
Yeah, the more I think about it, I think that's probably a good call. I was trying to avoid  returning an error, but I think the fact that you can technically provide any string, makes it important to handle that potentiality.

Reply:
Thank you both! Just implemented with returning an error. @DylanSpOddball , I like your suggestion about defining a new error type and using that specifically instead of fmt.ErrorOf. I'll add that now too and push up the update.